### PR TITLE
Fix real client name

### PR DESCRIPTION
**Pull request by NikoCat233**
Mod clients cant get real client name under current protocol, nobody is passing realname to them
So realclientname => AllClientRealNames and are synced by rpc, set by host at check name

Also there is no need of ref playername since all the vanilla set name steps  is done before the postfix
Removed the ref and do another rpc set name there

### CODE REVIEW

File: Patches/onGameStartedPatch.cs [hunk 1]
```diff
@@ -106,6 +106,7 @@ public static void Postfix(AmongUsClient __instance)
 
             // Sync Player Names
             RPC.SyncAllPlayerNames();
+            RPC.SyncAllClientRealNames();
```
[the remark this answers is not in the document]
Comment (NikoCat233) [reply Patches/onGameStartedPatch.cs:109]: makes sense

[the remark this answers is not in the document]
Comment (NikoCat233) [reply Patches/onGameStartedPatch.cs:109]: but real client name uses client id

[the remark this answers is not in the document]
Comment (Tommy-XL) [reply Patches/onGameStartedPatch.cs:109]: Just use

       writer.WritePacked(Main.AllPlayerNames.Count);
        foreach (var name in Main.AllPlayerNames)
        {
            writer.Write(name.Key);
            writer.Write(name.Value);
        }
        writer.WritePacked(Main.AllClientRealNames.Count);
        foreach (var name in Main.AllClientRealNames)
        {
            writer.Write(name.Key);
            writer.Write(name.Value);
        }

[the remark this answers is not in the document]
Comment (NikoCat233) [reply Patches/onGameStartedPatch.cs:109]: i already did

[the remark this answers is not in the document]
Comment (NikoCat233) [reply Patches/onGameStartedPatch.cs:109]: I some how cant boot my game

**Pull request merged**